browser/src/libs/code_intelligence/code_intelligence.tsx

```diff
@@ -259,47 +260,53 @@ export interface CodeHost extends ApplyLinkPreviewOptions {
     codeViewsRequireTokenization?: boolean
 }
 
+/**
+ * A blob (single file `FileInfo`) or a diff (with a head `FileInfo` and/or base `FileInfo`)
+ */
+export type DiffOrBlobInfo<T extends FileInfo = FileInfo> = ({ type: 'blob' } & T) | FileDiff<T>
+
+export type DiffOrBlobInfoOrError<T extends FileInfo = FileInfo> =
+    | DiffOrBlobInfo<T>
+    | { type: 'error'; error: ErrorLike }
+
+export type FileDiff<T extends FileInfo = FileInfo> = AddedFileDiff<T> | ModifiedFileDiff<T> | RemovedFileDiff<T>
+
+export interface AddedFileDiff<T extends FileInfo = FileInfo> {
+    type: 'added'
+    head: T
+}
+export interface ModifiedFileDiff<T extends FileInfo = FileInfo> {
+    type: 'modified'
+    head: T
+    base: T
+}
+export interface RemovedFileDiff<T extends FileInfo = FileInfo> {
+    type: 'removed'
+    base: T
+}
+
 export interface FileInfo {
     /**
-     * The path for the repo the file belongs to. If a `baseRepoName` is provided, this value
-     * is treated as the head repo name.
+     * The path for the repo the file belongs to.
      */
     rawRepoName: string
     /**
-     * The path for the file path for a given `codeView`. If a `baseFilePath` is provided, this value
-     * is treated as the head file path.
+     * The path for the file path for a given `codeView`.
      */
     filePath: string
     /**
-     * The commit that the code view is at. If a `baseCommitID` is provided, this value is treated
-     * as the head commit ID.
+     * The commit that the code view is at.
      */
     commitID: string
     /**
-     * The revision the code view is at. If a `baseRev` is provided, this value is treated as the head rev.
+     * The revision the code view is at.
      */
     rev?: string
-    /**
-     * The repo name for the BASE side of a diff. This is useful for Phabricator
-     * staging areas since they are separate repos.
-     */
-    baseRawRepoName?: string
-    /**
-     * The base file path.
-     */
-    baseFilePath?: string
-    /**
-     * Commit ID for the BASE side of the diff.
-     */
-    baseCommitID?: string
-    /**
-     * Revision for the BASE side of the diff.
-     */
-    baseRev?: string
 }
 
-export interface FileInfoWithRepoNames extends FileInfo, RepoSpec {
-    baseRepoName?: string
+export interface FileInfoWithRepoNames extends FileInfo, RepoSpec {}
```

Comment:
Just one repo name, right?
```suggestion
export interface FileInfoWithRepoName extends FileInfo, RepoSpec {}
```

Reply:
Thanks, good find!